hack/e2e-suite/guestbook.sh

```diff
@@ -38,30 +38,78 @@ GUESTBOOK="${KUBE_ROOT}/examples/guestbook"
 function teardown() {
   ${KUBECTL} stop -f "${GUESTBOOK}"
   if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
-    gcloud compute forwarding-rules delete "${INSTANCE_PREFIX}-default-frontend" || true
-    gcloud compute target-pools delete "${INSTANCE_PREFIX}-default-frontend" || true
+    gcloud compute forwarding-rules delete "${INSTANCE_PREFIX}-default-frontend" --region="${ZONE}" -q || true
+    gcloud compute target-pools delete "${INSTANCE_PREFIX}-default-frontend" --region="${ZONE}" -q || true
+    gcloud compute firewall-rules delete guestbook-e2e-minion-8000 -q || true
   fi  
 }
 
+function wait_for_running() {
+  echo "Waiting for pods to come up."
+  frontends=$(${KUBECTL} get pods -l name=frontend -o template '--template={{range.items}}{{.id}} {{end}}')
+  master=$(${KUBECTL} get pods -l name=redis-master -o template '--template={{range.items}}{{.id}} {{end}}')
+  slaves=$(${KUBECTL} get pods -l name=redisslave -o template '--template={{range.items}}{{.id}} {{end}}')
+  pods=$(echo $frontends $master $slaves)
+
+  all_running=0
+  for i in $(seq 1 30); do
+    sleep 10
+    all_running=1
+    for pod in $pods; do
+      status=$(${KUBECTL} get pods $pod -o template '--template={{.currentState.status}}') || true
+      if [[ "$status" == "Pending" ]]; then
+        all_running=0
+        break
+      fi
+    done
+    if [[ "${all_running}" == 1 ]]; then
+      break
+    fi
+  done
+  if [[ "${all_running}" == 0 ]]; then
+    echo "Pods did not come up in time"
+    exit 1
+  fi
+}
+
 prepare-e2e
 
 trap "teardown" EXIT
 
-echo "WARNING: this test is a no op that only attempts to launch guestbook resources."
 # Launch the guestbook example
 ${KUBECTL} create -f "${GUESTBOOK}"
 
-sleep 15
-
-POD_LIST_1=$(${KUBECTL} get pods -o template '--template={{range.items}}{{.id}} {{end}}')
-echo "Pods running: ${POD_LIST_1}"
-
-# TODO make this an actual test. Open up a firewall and use curl to post and
-# read a message via the frontend
-
-${KUBECTL} stop -f "${GUESTBOOK}"
-
-POD_LIST_2=$(${KUBECTL} get pods -o template '--template={{range.items}}{{.id}} {{end}}')
-echo "Pods running after shutdown: ${POD_LIST_2}"
+# Verify that all pods are running
+wait_for_running
+
+if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
+  gcloud compute firewall-rules create --allow=tcp:8000 --network=${NETWORK} --target-tags=${MINION_TAG} guestbook-e2e-minion-8000
+fi
+
+# Add a new entry to the guestbook and verify that it was remembered
+FRONTEND_ADDR=$(${KUBECTL} get service frontend -o template '--template={{range .publicIPs}}{{.}}{{end}}:{{.port}}')
+echo "Waiting for frontend to serve content"
+serving=0
+for i in $(seq 1 12); do
+  ENTRY=$(curl ${FRONTEND_ADDR}/index.php?cmd=get\&key=messages) || true
```

Comment:
Double quotes around the whole URL I'd say. Do you need a `http://` or similar protocol selector here?